sys/include/net/gnrc/ipv6/nib.h

```diff
@@ -391,12 +391,25 @@ void gnrc_ipv6_nib_handle_timer_event(void *ctx, uint16_t type);
  *                      router.
  */
 void gnrc_ipv6_nib_change_rtr_adv_iface(gnrc_netif_t *netif, bool enable);
+
+/**
+ * @brief   Immediately send a 'last router advertisement', that is a RA that
+ *          does not advertise this host as a default router.
+ *
+ *          If the @ref CONFIG_GNRC_IPV6_NIB_ADD_RIO_IN_LAST_RA option is set,
+ *          this will include a Route Information Option and nothing else.
+ *
+ * @param[in] netif     The interface on which the router advertisement should
+ *                      be sent.
+ */
+void gnrc_ipv6_nib_send_final_rtr_adv(gnrc_netif_t *netif);
```

Comment:
Why is this function needed? Shouldn't it be enough to disable `RTR_ADV` on the interface?

Reply:
And if not: one should rather use the `gnrc_ndp` function for that. The module for sending NDP messages is `gnrc_ndp`, `gnrc_ipv6_nib` is just a handler for these message types (and thus also sends them).

Reply:
Unfortunately not. `gnrc_ipv6_nib_change_rtr_adv_iface()` will just schedule the sending of the RA, if we did just send one it can take a bit.

That means if the upstream router was restarted for some reason it can take a while for routing to the downstream network to work again.

Reply:
In that case https://github.com/RIOT-OS/RIOT/pull/16536#discussion_r682475480 still applies.